frontend/src/pages/SharePage/SharePage.jsx

```diff
@@ -0,0 +1,192 @@
+                })
+                .catch((error) => {
+                    setError(error.message);
+                    setLoading(false);
+                });
+        } else {
+            setError("No encoded data found in URL.");
+            setLoading(false);
+        }
+    }, [encodedData]);
+
+    const handleGenerate = (dataParam) => {
+        if (
+            !dataParam.currUser ||
+            !dataParam.timeframe ||
+            !dataParam.selectedDemographic ||
+            !dataParam.selectedValues
+        ) {
+            console.warn(
+                "currUser or selectedDemographic is missing. Cannot generate data."
+            );
+            return;
+        }
+
+        if (!dataParam.selectedDemographic || dataParam.selectedValues[0] === "") {
+            alert("Upload data and model and select a demographic and values to generate data.");
+            return;
+        }
+
+        const fetchData = async (retries = 5, delay = 1000) => {
+            try {
+                const response = await axios.post(`${VITE_BACKEND_URL}/api/generate`, {
+                    demographics: [dataParam.selectedDemographic, dataParam.secondSelectedDemographic],
+                    choices: {
+                        [dataParam.selectedDemographic]: dataParam.selectedValues,
+                        [dataParam.secondSelectedDemographic]: dataParam.selectedSecondValues,
+                    },
+                    curr_user: dataParam.currUser,
+                    time: dataParam.timeframe,
+                });
+
+                if (response.data.length === 0) {
+                    alert("No data found for the selected demographics and values. Choose a different combination.");
+                    return;
+                }
+                setGraphData(response.data); // Set the graph data from the response
+                calculateAverageBias(response.data); // Calculate the average bias
+            } catch (err) {
+                console.error("Error generating data:", err);
+                // Retry logic: only retry if there are remaining attempts
+                if (retries > 0) {
+                    console.log(`Retrying... Attempts left: ${retries}`);
+                    setTimeout(() => fetchData(retries - 1, delay * 2), delay); // Exponential backoff
+                }
+            }
+        };
+
+        fetchData(); // Start the fetch process
+    };
+
+    const calculateAverageBias = (data) => {
+        if (!Array.isArray(data) || data.length === 0) {
+            throw new Error("Invalid data: Input must be a non-empty array.");
+        }
+
+        // Sum all the accuracy values
+        const totalAccuracy = data.reduce((sum, item) => sum + item.accuracy, 0);
+
+        // Calculate the average
+        const averageAccuracy = totalAccuracy / data.length;
+
+        setAverageBias(parseFloat(averageAccuracy.toFixed(2)));
+    };
+
+    const maxValue = () => {
```

Comment:
what's the purpose of this?

Reply:
This is for finding the maximum bias in the graph as this needs to be passed onto the chart component to decide which bar should be highlighted in red and which shouldn't. It is an indication to the engineers if their bias threshold is crossed or not. This value is being used by the slider component as well.